device/common/include/traccc/clusterization/device/impl/aggregate_cluster.ipp

```diff
@@ -70,32 +70,30 @@ inline void aggregate_cluster(
          * Terminate the process earlier if we have reached a cell sufficiently
          * in a different module.
          */
-        if (cells[pos].module_link != module_link) {
+        if (cells.module_index().at(pos) != module_idx) {
             break;
         }
 
-        const cell this_cell = cells[pos];
-
         /*
          * If the value of this cell is equal to our, that means it
          * is part of our cluster. In that case, we take its values
          * for position and add them to our accumulators.
          */
         if (f.at(j) == cid) {
 
-            if (this_cell.channel1 > maxChannel1) {
-                maxChannel1 = this_cell.channel1;
+            if (cells.channel1().at(pos) > maxChannel1) {
+                maxChannel1 = cells.channel1().at(pos);
```

Comment:
General comment: please replace all the use of array indexing in this file with proxy syntax.

Reply:
No. I really-really don't intend to. :thinking:

Reply:
I ended up doing sort of this.